extensions/src/test/java/org/pentaho/platform/web/http/api/resources/EmailResourceTest.java

```diff
@@ -93,7 +96,28 @@ public void testEmailConfig() throws Exception {
   }
 
   public void testSendEmailTest() throws Exception {
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "test@pentaho.com", "Pentaho Scheduler", "", 25,
+                    "graph_api", true, "user", "", true, true );
+    emailConfigOriginal.setAuthMechanism( "XOAUTH2" );
+    emailConfigOriginal.setClientId( "cid1" );
+    emailConfigOriginal.setClientSecret( "cid1111" );
+    emailConfigOriginal.setTokenUrl( "https://login.microsoftonline.com/common/oauth2/v2.0/token" );
+    emailConfigOriginal.setScope( "offline_access mail.send" );
+    emailConfigOriginal.setRefreshToken( "token" );
+    emailConfigOriginal.setGrantType( "refresh_token" );
+    emailResource.sendEmailTest( emailConfigOriginal );
+  }
 
+  public void testSendEmailTestBasicAuth() throws Exception {
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "test@pentaho.com", "Pentaho Scheduler", "smtp.com", 25,
+                    "SMTP", true, "user", "password", true, true );
+    emailConfigOriginal.setAuthMechanism( "basic" );
+    try ( MockedStatic<Encr> encr = Mockito.mockStatic( Encr.class ) ) {
+      encr.when( () -> Encr.encryptPasswordIfNotUsingVariables( emailConfigOriginal.getPassword() ) ).thenReturn( emailConfigOriginal.getPassword() );
```

Comment:
The previous author(s) didn't limit their tests strictly to the class under test `EmailResource`. This is why they are using real implementation of third party classes.

The EmailResourceTest should look something like this: https://github.com/NJtwentyone/pentaho-platform/tree/poc/BACKLOG-36727/EmailResourceTest-sendTestEmail

There is still some work to do.